akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -886,7 +910,13 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       // removing REMOVED nodes from the `reachability` table
       val newReachability = localOverview.reachability.remove(removed)
       val newOverview = localOverview copy (seen = newSeen, reachability = newReachability)
-      val newGossip = localGossip copy (members = newMembers, overview = newOverview)
+      val newVersion = removed.foldLeft(localGossip.version) { (v, node) ⇒
+        v.prune(VectorClock.Node(vclockName(node)))
+      }
+      val newGossip = localGossip copy (members = newMembers, overview = newOverview, version = newVersion)
```

Comment:
I see. So it is the leader that removes the entry, and this removal is then propagated by the others. This is a safe solution since the leader is always unique (but might not exist if there is a nonconvergent period).

So the only non-monotonic decision is the first removal, which is concentrated to the leader, the rest of the removal is handled by the gossip merge in a monotonic, eventually consistent fashion.